panel_grid.py

```diff
@@ -46,34 +55,38 @@ def create_panel(self):
         tab = Frame(tablayout)
         tab.pack(fill="both")
 
-        for i in range(6):              # for rows in activities
-            for j in range(8):          # for cols in days
+        for i in range(len(self.activities[self.week-1])+1):              # for rows in activities
+            for j in range(len(self.days[self.week-1])+1):          # for cols in days
                 if i == 0 and j == 0:                                  # if its first cell, add empty cell
                     label = Label(tab, text=" ")
                     label.grid(row=i, column=j)                        # this specifies where in the grid
                     tab.grid_columnconfigure(j, weight=1)              
                     # this last line makes the width of the column responsive to change in width of the window
                 elif i == 0:           # adding the name of the day
-                    label = Label(tab, text=self.days[j-1])
+                    label = Label(tab, text=self.days[self.week-1][j-1])
                     label.grid(row=i, column=j)
                     tab.grid_columnconfigure(j, weight=1)
                 elif j == 0:           # adding the name of the activity
-                    label = Label(tab, text=self.activities[i-1])
+                    label = Label(tab, text=self.activities[self.week-1][i-1])
                     label.grid(row=i, column=j)
                     tab.grid_columnconfigure(j, weight=1)
                 else:                  # adding the checkboxes
                     label = Checkbutton(tab) 
                     label.grid(row=i, column=j)
                     tab.grid_columnconfigure(j, weight=1)
         
-        # Oh shit there is no pack in this for loop at all, yeah man never
+        he = Button(tab,text='hello',command=self.add_week)
```

Comment:
You see here the button need to be in it's own frame, like I said in the function definition, you should play with that Frame() to understand it.